msg.go

```diff
@@ -852,16 +852,25 @@ func (dns *Msg) unpack(dh Header, msg []byte, off int) (err error) {
 	}
 
 	dns.Answer, off, err = unpackRRslice(int(dh.Ancount), msg, off)
+	if err == nil && int(dh.Ancount) > len(dns.Answer) {
+		return ErrBuf
+	}
```

Comment:
Now we return ErrBuf and the code below is only executed when the count is smaller which is weird. So this def. feels wrong. And can't go in as such

Again: an upstream sends back a message that is too large without TC? That is a protocol violation, unless I still misunderstand

Reply:
Thank you very much for the reply - @miekg. Sure sure I understand. Yes, this is a scenario seen due to non-complying DNS server. 

@tmthrgd mentioned the fix is actually much simpler than what I proposed here https://github.com/miekg/dns/pull/1493#issuecomment-1802334548. I am kindly requesting Tom's help here. 

@rdrozhdzh has very nicely summarized the issue I am trying to explain - https://github.com/miekg/dns/pull/1493#issuecomment-1794532532. 

Quote - Sporadic "legitimization" of malformed DNS responses by dns library under some condition, i.e. the library doesn't return error, but instead it silently returns (possibly truncated or "fixed") DNS message.

The type of malformed message being considered here is when the number of resource records declared in the message header does not correspond to the actual number of resource records existing in DNS response message. As a special case, this may happen when dns library allocates smaller UDP buffer than the size of DNS response coming from (non-complying) DNS server.

The condition when the error is not reported by dns library is if the last octet of DNS message (i.e. the last byte of the buffer) corresponds to last octet of one of resource records.

Reply:
@tmthrgd - Earlier you had mentioned about a simpler fix for this issue, would you be able to help please ?